src/Runner/Runner.php

```diff
@@ -105,10 +127,47 @@ private function createProcessQueue(): void
         }
     }
 
-    public function pushToPipeline(): void
+    private function createChunkedProcessQueue(): void
     {
+        $files = $this->filter->filterTestFiles();
+        foreach (array_chunk($files, $this->chunkSize->getChunkSize()) as $chunkNumber => $filesChunk) {
```

Comment:
Would it be possible to create chunks by a custom function and have it as an extension point? For a use-case: some of our tests are also expensive to bootstrap, but in our case it is Symfony's container that needs to be rebuilt due to test's specifics. It would be way better to have them executed by a single process (or at least same "worker" that did it before) as it already has right kind of cache warmed up.

I can of course help with implementation as I'm the one needing something. Just found this PR during problem analysis and thought I'll chime in here first as my problem is almost solved here :)

Reply:
That would be nice, but it would also be hard to find an easy extension point to attach to... If you have any suggestion try opening a new issue or continuing inside #163

Reply:
If I'm looking correctly (but without diving into code yet) this very line would be a perfect extension point. The `array_chunk` would be delegated to a class implementing let's say, `TestChunkerInterface`. Said service's responsibility would be to chunk given tests into whatever it sees appropriate :) 

I'll try to get back with some code this week, posting this in case I'd be dead wrong

Reply:
@malarzm maybe you could wrap this processing in PHPUnit's custom bootstrap?

`<?xml version="1.0"?>
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd" 
         backupGlobals="false" 
         cacheResult="false"
         colors="true"
         **bootstrap="vendor/autoload.php"**
         timeoutForSmallTests="1"
         timeoutForMediumTests="5"
         timeoutForLargeTests="10"
         beStrictAboutOutputDuringTests="true"
>`

Reply:
@pczerkas could you elaborate a bit on the idea?

To give a broader picture, we're operating in a semi-saas setup We have one main environment variable designated as a tenant's identifier which configures the app and its DI container in a certain way. Our acceptance tests are developed in a way that lets them change the store identifier on demand, which then requires Symfony to realize it has no container for a given setup (or maybe it has one from a previous execution, then nothing needs to be done). So all in all one "worker" can execute tests for multiple tenants and regenerating cache for one will not invalidate cache for another. Current idea is that if we can have one "worker" execute tests for all tenant's tests, the cache will be warmed up once (and only once), not multiple times (once per worker executing of tests, pessimistically we are warming up cache `n` times where `n` is number of workers). That's what I'll be willing to achieve with a custom chunker - chunk the tests so single tenant's tests will be grouped together. Such tests make up for ~10% of the test suite, where we really need to test tenant's configured behaviour, the rest can be tested in a tenant-agnostic way so I'm fine chunking the rest in whatever way